receiver/jaegerreceiver/internal/processor/thrift_processor.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package processor
+
+import (
+	"context"
+	"sync"
+
+	apacheThrift "github.com/apache/thrift/lib/go/thrift"
+	"github.com/jaegertracing/jaeger-idl/thrift-gen/agent"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver/internal/udpserver"
+)
+
+// ThriftProcessor is a processor that handles Thrift over UDP
+type ThriftProcessor struct {
+	server          *udpserver.UDPServer
+	protocolFactory apacheThrift.TProtocolFactory
+	handler         agent.Agent
+	logger          *zap.Logger
+	stopCh          chan struct{}
+	stopOnce        sync.Once
+}
+
+// NewThriftProcessor creates a new ThriftProcessor
+func NewThriftProcessor(
+	endpoint string,
+	maxPacketSize int,
+	protocolFactory apacheThrift.TProtocolFactory,
+	handler agent.Agent,
+	logger *zap.Logger,
+) (*ThriftProcessor, error) {
+	processor := &ThriftProcessor{
+		protocolFactory: protocolFactory,
+		handler:         handler,
+		logger:          logger,
+		stopCh:          make(chan struct{}),
+	}
+
+	// Create a packet handler that will process the UDP packets
+	packetHandler := &thriftPacketHandler{
+		protocolFactory: protocolFactory,
+		handler:         handler,
+		logger:          logger,
+	}
+
+	// Create the UDP server
+	server, err := udpserver.New(endpoint, maxPacketSize, packetHandler, logger)
+	if err != nil {
+		return nil, err
+	}
+	processor.server = server
+
+	return processor, nil
+}
+
+// Start starts the processor
+func (p *ThriftProcessor) Start() error {
+	return p.server.Start()
+}
+
+// Stop stops the processor
+func (p *ThriftProcessor) Stop() error {
+	p.stopOnce.Do(func() {
+		close(p.stopCh)
+		p.server.Stop()
+	})
+	return nil
+}
+
+// Done returns a channel that's closed when the processor has stopped
+func (p *ThriftProcessor) Done() <-chan struct{} {
```

Comment:
Conceptually Thrift processor is just a handler that you need to pass to UDP Server. I recommend using more traditional setup for receivers
  * receiver creates Thrift Handler which is stateless, it only transforms the packet and hands it off to the next trace consumer in the pipeline. Thrift Handler needs no start/stop.
  * receiver creates UDP server which has start/stop and runs in a goroutine. The Thrift Handler is passed to server at construction